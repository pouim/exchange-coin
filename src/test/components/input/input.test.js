import { shallow } from 'enzyme';
import Input from 'src/components/ui/input';

describe('Input', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Input debug />);

        expect(component).toMatchSnapshot();
    });

    it('should render placeholder correctly', () => {
        const component = shallow(<Input placeholder="test" />);
        const placeholderContainer = component.find('[data-test="custom-input-placeholder"]');

        expect(placeholderContainer.text()).toBe('test');
    });

    it('should render error message correctly', () => {
        const testError = {
            message: 'test error message',
        };

        const component = shallow(<Input error={testError} />);
        const placeholderContainer = component.find('[data-test="custom-input-error"]');

        expect(placeholderContainer).toHaveLength(1);
        expect(placeholderContainer.text()).toBe('test error message');
    });
});
