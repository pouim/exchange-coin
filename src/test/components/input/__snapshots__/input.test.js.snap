// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`MyComponent should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
